back_end/README.md

```diff
@@ -1,21 +1,83 @@
+- [ ] As a writer I should be able to edit any of my articles
+- [ ] As a writer I should be able to list all my articles (published and drafted state)
+- [ ] As a writer I should be able to see a list of only my drafted articles or only my published articles
+- [ ] As a writer I should be able to see one article
+- [ ] As a writer I should be able to do everything a visitor can do in the API
+- [ ] As a visitor I should be able to list all the published articles order by published date in descending order
+- [ ] As a visitor I should be able to see a published article 
 
 ## Requirements
 
-- [ ] Create a webservice using Java, Ruby, or Node.js that has a single endpoint.
-- [ ] The endpoint should receive a URL as a parameter and return a shortened URL. For example, if the endpoint is given the following URL: "https://facebook.com", it should return "https://hostname/Ux26Yp".
-- [ ] When someone accesses the shortened URL ("https://hostname/Ux26Yp"), the server should redirect to the real URL ("https://facebook.com").
-- [ ] Add another endpoint to your service that allows for the bulk (hundreds) submission of URLs.
-- [ ] Your submission should contain a README file that spells out the exact steps necessary to build and run the project. It should also provide instructions for how to query the webservice.
+- [ ] You can select the language you think is the best for this challenge. If you decide to use a functional programming like elixir just let us know first.
+- [ ] You are free to add any library for improving the development experience, like, linters, formatters, etc. Try to emulate the environment that you would like to have when you are working.
+- [ ] You should design your APi for evolvability. Given that in real life no API stays static for long.  
+- [ ] You can use any API paradigm. Rest, RPC or GraphQL depending on what you think is best for the challenge. 
+- [ ] The app should have the tests that you decide are necessary for maintainability
+- [ ] You should handle all success and errors cases the endpoints could have. When there is an error the backend should notify the client about it in such a way that allows the client side to decide what to do. (Ex. If the client tries to create an article without a title we expect the API to notify the user that the title is a required attribute in order to create an article) 
+- [ ] There should be a way for client users to discover how to use the API. It can be a document, a postman, a swagger document or a plain html website. Is up to you but it should allow the user to discover all your endpoints, and their possible responses both success and errors. 
 
-Design of the API endpoint including paths, structure, and format is up to you. Decisions on how to structure the code and what frameworks to use are up to you as well.
+## Bonus Points 
 
-## Bonus
+Take 1 or 2, we don't expect for you to do everyone of them. 
```

Comment:
I think this `everyone` should be separated. `"we don't expect for you to do every one of them"` but I'm not that sure. 🤔